src/ImageView.jl

```diff
@@ -476,11 +476,11 @@ function histsignals(enabled::Signal, defaultimg, img::Signal, clim::Signal{CLim
     return histsigs
 end
 
-function scaleminmax(mtyp::DataType, cmin::AbstractRGB{T}, cmax::AbstractRGB{T}) where {T}
+function scaleminmax(::Type{Tout}, cmin::AbstractRGB{T}, cmax::AbstractRGB{T}) where {T,Tout}
```

Comment:
This change makes this function inferrable. Since it runs on every pixel, that's important. A `DataType` type is typically a recipe for non-inferrability.

Reply:
I think this brings out a longstanding point of confusion for me.  Kind of amazing that I still don't get this right.  I thought that annotating `mtyp` in this case was only for readability / catching errors and did not affect inferrability.  I thought that as long as the compiler can infer the type of `mtyp` and all operations within the `scaleminmax` are themselves type stable then the entire `scaleminmax` function should be type stable.  But am I correct that data types are fundamentally non-inferrable and for this reason the `Type{xxx}` construct was invented to wrap a datatype and make it inferrable?

Reply:
Right, types-as-arguments are in a bit of a special category. You can think of arguments as tuples, and type-elements of tuples "lose" their type info:

```julia
julia> typeof((1,"hello",Int))
Tuple{Int64,String,DataType}
```

Reply:
Ah this makes sense now, thanks!  And it's helpful to know that args are just Tuples in case I forget this again.